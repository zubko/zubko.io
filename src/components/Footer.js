/**
 * Common footer of the website
 *
 * @flow
 */

import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faReact } from '@fortawesome/free-brands-svg-icons';

import { Link } from '../components';
import { rhythm } from '../Typography';

export default () => (
  <footer css={{ marginTop: rhythm(1) }}>
    <p>
      ~~~
      <br /> Â© {new Date().getFullYear()}, Built{' '}
      <Link css={Styles.link} usePlainStyle to="/acknowledgements">
        with ðŸ’œ and open source{' '}
        <FontAwesomeIcon
          icon={faReact}
          css={Styles.faIcon}
          size="lg"
          width={25}
          height={25}
        />
      </Link>
    </p>
  </footer>
);

const Styles = {
  link: {
    textDecoration: 'none',
    background: 'none',
  },
  faIcon: { color: 'violet' },
};
