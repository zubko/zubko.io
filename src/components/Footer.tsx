/**
 * Common footer of the website
 */

import { faReact } from '@fortawesome/free-brands-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { Link } from '../components';
import { rhythm } from '../Typography';

export default () => (
  <footer css={{ marginTop: rhythm(1) }}>
    <p>
      ~~~
      <br /> Â© {new Date().getFullYear()}, Built{' '}
      <Link css={Styles.link} usePlainStyle to="/acknowledgements">
        with ðŸ’œ and open source{' '}
        <FontAwesomeIcon
          icon={faReact}
          css={Styles.faIcon}
          size="lg"
          width={25}
          height={25}
        />
      </Link>
    </p>
  </footer>
);

const Styles = {
  link: {
    textDecoration: 'none',
    background: 'none',
  },
  faIcon: { color: 'violet' },
};
