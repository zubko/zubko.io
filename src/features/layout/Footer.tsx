/**
 * Common footer of the website
 */

import { faReact } from "@fortawesome/free-brands-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styled from "styled-components";

import { Link } from "../../components/Link";
import { rhythm } from "../../Typography";

export const Footer = () => (
  <Container>
    <p>
      ~~~
      <br /> Â© {new Date().getFullYear()}, Built{" "}
      <AcknowledgementsLink usePlainStyle to="/acknowledgements">
        with ðŸ’œ and open source{" "}
        <Icon icon={faReact} size="lg" width={25} height={25} />
      </AcknowledgementsLink>
    </p>
  </Container>
);

const Container = styled.footer`
  margin-top: ${rhythm(1)};
`;

const AcknowledgementsLink = styled(Link)`
  text-decoration: none;
  background: none;
`;

const Icon = styled(FontAwesomeIcon)`
  color: violet;
`;
