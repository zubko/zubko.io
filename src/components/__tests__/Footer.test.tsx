import React from "react";
import renderer from "react-test-renderer";

import Footer from "../Footer";

describe("Footer", () => {
  it("renders without regressions", () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
