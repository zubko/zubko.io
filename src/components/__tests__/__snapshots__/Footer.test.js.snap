// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders without regressions 1`] = `
<footer
  className="css-1o3tyej"
>
  <p>
    ~~~
    <br />
     © 
    2019
    , Built
     
    <a
      className="css-1a4dihx-linkPlain"
      href="/acknowledgements"
    >
      with ❤️and open source ⚛
    </a>
  </p>
</footer>
`;
